src/scheduler.cc

```diff
@@ -127,6 +127,16 @@ Status SchedulerService::RegisterWorker(ServerContext* context, const RegisterWo
   return Status::OK;
 }
 
+Status SchedulerService::WorkerReady(ServerContext* context, const WorkerReadyRequest* request, AckReply* reply) {
+  WorkerId workerid = request->workerid();
+  RAY_LOG(RAY_INFO, "worker with workerid " << workerid << " can now receive tasks");
+  avail_workers_lock_.lock();
```

Comment:
Never call lock directly when it is scoped in the same function. Use `std::lock_guard`.